mod-di-converter-storage-server/src/main/resources/templates/db_scripts/associations-migration-gen-table/associations_migration.sql

```diff
@@ -0,0 +1,322 @@
+            (r.jsonb ->> 'detailProfileId')::uuid,
+           'JOB_PROFILE',
+           'ACTION_PROFILE',
+            (r.jsonb ->> 'order')::int,
+            null
+           ) ON CONFLICT DO NOTHING;
+
+      END LOOP;
+    RAISE NOTICE 'PROFILES_MIGRATION:: migrated from job_to_action_profiles';
+  END
+$$;
+
+DO
+-- match_to_match_profiles: migration
+$$
+  DECLARE
+    not_recursive_record    record;
+    recursive_record        record;
+    detail_match_wrapper_id UUID;
+    master_match_wrapper_id UUID;
+  BEGIN
+    FOR not_recursive_record IN
+      -- get match profiles that are associated with job profile. Each item in the result will be used to build a match hierarchy
+      select jtm.id, jtm.jsonb
+      from job_to_match_profiles jtm
+      LOOP
+        FOR recursive_record IN
+          WITH RECURSIVE match_hierarchy AS (
+            -- Base Case
+            SELECT mtm.id, mtm.jsonb
+            FROM match_to_match_profiles mtm
+            where mtm.jsonb ->> 'masterProfileId' = not_recursive_record.jsonb ->> 'detailProfileId'
+              and mtm.jsonb ->> 'jobProfileId' = not_recursive_record.jsonb ->> 'masterProfileId'
+            UNION ALL
+            -- Recursive Step
+            SELECT m.id, m.jsonb
+            FROM match_to_match_profiles m
+                   INNER JOIN match_hierarchy mh
+                              ON mh.jsonb ->> 'detailProfileId' = m.jsonb ->> 'masterProfileId'
+                                and mh.jsonb ->> 'jobProfileId' = m.jsonb ->> 'jobProfileId')
+
+          select *
+          from match_hierarchy
+          LOOP
+
+            -- get existing wrapper for master match profile
+            select id
+            into master_match_wrapper_id
+            from profile_wrappers
+            where match_profile_id = (recursive_record.jsonb ->> 'masterProfileId')::uuid
+              and associated_job_profile_id = (recursive_record.jsonb ->> 'jobProfileId')::uuid;
+
+            -- insert into new association table
+            INSERT INTO profile_associations (id, job_profile_id, master_wrapper_id,
+                detail_wrapper_id, master_profile_id, detail_profile_id,
+                master_profile_type, detail_profile_type, detail_order, react_to) values
+                (recursive_record.id,
+                (recursive_record.jsonb ->> 'jobProfileId')::uuid,
+                master_match_wrapper_id,
+                detail_match_wrapper_id,
+                (recursive_record.jsonb ->> 'masterProfileId')::uuid,
+                (recursive_record.jsonb ->> 'detailProfileId')::uuid,
+               'MATCH_PROFILE',
+               'MATCH_PROFILE',
+                (recursive_record.jsonb ->> 'order')::int,
+                (recursive_record.jsonb ->> 'reactTo')::text
+               ) ON CONFLICT DO NOTHING;
+
+          end loop;
+      end loop;
+    raise notice 'PROFILES_MIGRATION:: migrated from match_to_match_profiles';
+  end
+$$;
```

Comment:
```suggestion
INSERT INTO profile_associations (id, job_profile_id, master_wrapper_id,
        detail_wrapper_id, master_profile_id, detail_profile_id,
        master_profile_type, detail_profile_type, detail_order, react_to) values
        (jsonb.id,
        (jsonb ->> 'jobProfileId')::uuid,
        (jsonb ->> 'masterProfileId')::uuid,
        (jsonb ->> 'detailProfileId')::uuid,
        (jsonb ->> 'masterProfileId')::uuid,
        (jsonb ->> 'detailProfileId')::uuid,
        'MATCH_PROFILE',
        'MATCH_PROFILE',
        (jsonb ->> 'order')::int,
        (jsonb ->> 'reactTo')::text
       )
select jsonb
from match_to_match_profiles
```

it is not necessary to search in match profiles hierarchy. We just can/need to insert existing data from old associations tables into the new associations table. Also, no other additional checks are required

Reply:
yes, good point, also had concerns about this part, but the match-to-match relationship has a hierarchical structure? also with this proposed approach for me, it seems like we might lose association between parent and child? since we are passing detail and master profile id instead of wrappers?

Reply:
but anyway let me check it, to be sure

Reply:
I understand your concern. Sorry, I made a typo in the approach example. I specified masterProfileId and detailProfileId twice instead of masterWrapperId and detailWrapperId for the corresponding columns. Thanks for letting me know. I've updated the suggestion example with wrapperId, please take a look.

Reply:
The idea of the approach is that we simply move associations from the old table to the new table without additional overhead, as the associations in the old table already contain the necessary data.

Reply:
thank you Ruslan, for attention to the details, fixed it)